homeassistant/components/ruckus_unleashed/device_tracker.py

```diff
@@ -0,0 +1,105 @@
+"""Support for Ruckus Unleashed devices."""
+from typing import Optional
+
+from homeassistant.components.device_tracker import SOURCE_TYPE_ROUTER
+from homeassistant.components.device_tracker.config_entry import ScannerEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_NAME
+from homeassistant.core import callback
+from homeassistant.helpers import entity_registry
+from homeassistant.helpers.typing import HomeAssistantType
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import CLIENTS, COORDINATOR, DOMAIN, UNDO_UPDATE_LISTENERS
+
+
+async def async_setup_entry(
+    hass: HomeAssistantType, entry: ConfigEntry, async_add_entities
+) -> None:
+    """Set up device tracker for Ruckus Unleashed component."""
+    coordinator = hass.data[DOMAIN][entry.entry_id][COORDINATOR]
+
+    tracked = set()
+
+    def router_update():
+        """Update the values of the router."""
+        add_new_entities(coordinator, async_add_entities, tracked)
+
+    router_update()
+
+    hass.data[DOMAIN][entry.entry_id][UNDO_UPDATE_LISTENERS].append(
+        coordinator.async_add_listener(router_update)
+    )
+
+    registry = await entity_registry.async_get_registry(hass)
+    restore_entities(registry, coordinator, entry, async_add_entities, tracked)
```

Comment:
We restore already all entities on startup. Why you need this?

Reply:
Hm I knew that entities should to restore on startup, but I was having issues where they would show up with a red icon and would be unavailable. That may have been fixed as I cleaned up the actual `RuckusUnleashedDevice` class so I'll remove that and see if it works any better now. Thank you!

Reply:
I wasn't aware of the `RestoreEntity` class. 🤦  Removing the custom logic now. Thanks for pointing me in the right direction!

Reply:
Actually nevermind, `RestoreEntity` won't work how I am needing. The device does not show up in the router response at all if not connected so it never hits `async_added_to_hass`. The current `restore_entities` function is very similar to how the `unifi` and the `mikrotik` integrations add missing clients back after a restart. Is there something I'm missing here?